lib/core.ml

```diff
@@ -627,28 +633,33 @@ let rec translate ids kind ign_group ign_case greedy pos cache c = function
     (A.before ids Category.search_boundary, kind)
   | Sem (kind', r') ->
     let (cr, kind'') =
-      translate ids kind' ign_group ign_case greedy pos cache c r' in
+      translate ids kind' ign_group ign_case greedy pos names cache c r' in
     (enforce_kind ids kind' kind'' cr,
      kind')
   | Sem_greedy (greedy', r') ->
-    translate ids kind ign_group ign_case greedy' pos cache c r'
-  | Group r' ->
+    translate ids kind ign_group ign_case greedy' pos names cache c r'
+  | Group (n, r') ->
     if ign_group then
-      translate ids kind ign_group ign_case greedy pos cache c r'
+      translate ids kind ign_group ign_case greedy pos names cache c r'
     else
       let p = !pos in
+      let () =
+        match n with
+        | Some name -> names := (p / 2, name) :: !names
```

Comment:
Should we accept duplicated names ?

Reply:
PCRE has a mode where it accepts duplicate names, so I don't see the problem with us accepting them. Besides, they are totally usable (and useful) in my implementation.